docker-app/qfieldcloud/core/utils2/storage.py

```diff
@@ -96,3 +100,54 @@ def remove_project_thumbail(project: "Project") -> None:  # noqa: F821
     bucket = qfieldcloud.core.utils.get_s3_bucket()
     key = project.thumbnail_uri
     bucket.object_versions.filter(Prefix=key).delete()
+
+
+def purge_old_file_versions(project: "Project") -> None:  # noqa: F821
+    """
+    Deletes old versions of all files in the given project. Will keep __3__
+    versions for COMMUNITY user accounts, and __10__ versions for PRO user
+    accounts
+    """
+
+    logger.info(f"Cleaning up old files for {project}")
+
+    # Determine account type
+    account_type = project.owner.useraccount.account_type
+    if account_type == UserAccount.TYPE_COMMUNITY:
+        keep_count = 3
+    elif account_type == UserAccount.TYPE_PRO:
+        keep_count = 10
+    else:
+        raise NotImplementedError(f"Unknown account type {account_type}")
+
+    logger.debug(f"Keeping {keep_count} versions")
+
+    # Process file by file
+    for file in qfieldcloud.core.utils.get_project_files_with_versions(project.pk):
+
+        filename = file.latest.name
+        old_versions = file.versions
+
+        # Sort by date (newest first)
+        old_versions.sort(key=lambda i: i.last_modified, reverse=True)
+
+        # Skip the newest N
+        old_versions_to_purge = old_versions[keep_count:]
+
+        # Debug print
+        all_count = len(old_versions)
+        topurge_count = len(old_versions_to_purge)
+        logger.debug(
+            f'Purging {topurge_count} out of {all_count} old versions for "{filename}"...'
+        )
```

Comment:
```suggestion
        logger.debug(
            f'Purging {len(old_versions_to_purge)} out of {len(old_versions)} '
            f'old versions for "{filename}"...'
        )
```

Sorry, I just hate extra variable mappings... I easily get confused by them.

Reply:
done